src/mon/Monitor.cc

```diff
@@ -3143,6 +3155,24 @@ void Monitor::handle_command(MonOpRequestRef op)
     rdata.append(ds);
     rs = "";
     r = 0;
+  } else if (prefix == "features") {
+    if (!is_leader() && !is_peon()) {
+      dout(10) << " waiting for quorum" << dendl;
+      waitfor_quorum.push_back(new C_RetryMessage(this, op));
+      return;
+    }
+    if (!is_leader()) {
+      forward_request_leader(op);
```

Comment:
i don't really understand why only the leader's featuremap matters?

the peon has its own `session_map` and keeps track of the quorum also. but i agree, its `session_map` is different from the leader's. the question is why we should use the leader's `session_map` but not the peon's?